assets/js/util/index.js

```diff
@@ -714,28 +714,49 @@ export const deleteCache = ( cacheType, cacheKey ) => {
 	return true;
 };
 
-export const findTagInIframeContent = ( iframe, module ) => {
+export const findTagInHtmlContent = ( html, module ) => {
 	let existingTag = false;
 
-	if ( ! iframe ) {
+	if ( ! html ) {
 		return false;
 	}
 
-	// Check if tag present in <head>.
-	const head = iframe.contentWindow.document.querySelector( 'head' );
-	if ( head ) {
-		existingTag = extractTag( head.innerHTML, module );
-	}
+	existingTag = extractTag( html, module );
+
+	return existingTag;
+};
+
+/**
+ * Looks for existing tag requesting front end html, if no existing tag was found on server side
+ * while requesting list of accounts.
+ *
+ * @param {string} module Module slug.
+ */
+export const getExistingTag = async( module ) => {
+
+	try {
+		let tagFound = data.getCache( module, 'existingTag', 300 );
+
+		if ( ! tagFound ) {
+			const html = await fetch( `${googlesitekit.admin.siteURL}?tagverify=1&timestamp=${Date.now()}` ).then( res => {
+				return res.text();
+			} );
 
-	// If not in <head> check if tag present in <body>.
-	if ( false === existingTag ) {
-		const body = iframe.contentWindow.document.querySelector( 'body' );
-		if ( body ) {
-			existingTag = extractTag( body.innerHTML, module );
+			tagFound = findTagInHtmlContent( html, module );
+			if ( ! tagFound ) {
+				tagFound = 'no tag';
```

Comment:
Why do we set the variable to this arbitrary string and not an empty string or `false`?

Reply:
Changing to empty string. Setting to false will be interpreted as cache value is false and will make the request over and over.

Reply:
@lucymtc These kind of details are important. Please add a comment for this kind of thing in the future as it isn't obvious to the reader when skimming. Thanks!